packages/walletkit/src/signing-utils.ts

```diff
@@ -1,12 +1,19 @@
 import { account as Account, bytes as Bytes, hash as Hash, nat as Nat, RLP } from 'eth-lib'
 import { extend, isNull, isUndefined } from 'lodash'
+import * as util from 'util'
 import * as helpers from 'web3-core-helpers'
 import * as utils from 'web3-utils'
 import { Logger } from './logger'
+import { generateAccountAddressFromPrivateKey } from './new-web3-utils'
+import { CeloPartialTxParams } from './transaction-utils'
 
 // Original code taken from
 // https://github.com/ethereum/web3.js/blob/1.x/packages/web3-eth-accounts/src/index.js
 
+// Debug-mode only: Turn this on to verify that the signing key matches the sender
+// before signing as well as the recovered signer matches the original signer.
+const DEBUG_MODE_CHECK_SIGNER = false
```

Comment:
Just curious, why only check in debug mode?

Reply:
It feels like a redundant expensive check.